src/main/java/frc/trigon/robot/subsystems/intake/Intake.java

```diff
@@ -35,6 +35,19 @@ public IntakeConstants.IntakeState getTargetState() {
         return targetState;
     }
 
+    public boolean hasNote() {
+        return IntakeConstants.HAS_NOTE_BOOLEAN_EVENT.getAsBoolean();
+    }
+
+    /**
+     * Checks if a note has been collected early, using the motor's current.
+     *
```

Comment:
Explain how this is different from hasNote, why is this faster, and where this should be used.